### PR TITLE
Feature/scoring results

### PR DESCRIPTION
**Pull request by Mironor**
Adds scoring in the api response for addresses search

### CODE REVIEW

File: model/src/test/scala/uk/gov/ons/addressIndex/model/AddressIndexModelTest.scala [hunk 1]
```diff
@@ -7,27 +7,32 @@ class AddressIndexModelTest extends FlatSpec with Matchers {
...
 
   it should "produce the correct `AddressScheme` (`BritishStandard7666`) for the given string `bs7666`" in {
     val sample = "bs7666"
-    sample.stringToScheme shouldBe a [BritishStandard7666]
+    sample.stringToScheme shouldBe Some(BritishStandard7666(sample))
   }
 
   it should "produce the correct `AddressScheme` (`BritishStandard7666`) for the given string `bs`" in {
     val sample = "bs"
-    sample.stringToScheme shouldBe a [BritishStandard7666]
```
Comment (iRhysBradbury) [inline model/src/test/scala/uk/gov/ons/addressIndex/model/AddressIndexModelTest.scala]: test name on line 33 needs changing

File: server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala [hunk 2]
```diff
@@ -63,29 +62,28 @@ class AddressController @Inject()(esRepo: ElasticsearchRepository)(implicit ec:
...
-          tokens,
-          addresses = addresses.map(AddressResponseAddress.fromPafAddress),
-          limit = 10,
-          offset = 0,
-          total = addresses.size
-        ),
-        AddressResponseStatus.ok,
-        errors = Seq()
-      )))
+    esRepo.queryAddress(tokens).map {
```
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: Non exhaustive match

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala [hunk 3]
```diff
@@ -95,5 +95,5 @@ class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticC
         )
       )
     }
-  }.map(_.as[PostcodeAddressFileAddress])
+  }.map(response => (response.as[PostcodeAddressFileAddress], response.maxScore))
 }
```
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: Maybe a case class describing what the Float is, as opposed to a tuple

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala [hunk 4]
```diff
@@ -80,7 +80,7 @@ class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticC
     search in pafIndex query { termQuery("uprn", uprn) }
   }.map(_.as[PostcodeAddressFileAddress].headOption)
```
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: scaladoc

File: server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec.scala [hunk 5]
```diff
@@ -121,15 +121,16 @@ class ElasticsearchRepositorySpec extends WordSpec with SearchMatchers with Elas
...
       // When
-      val result = repository.queryAddress(tokens).await
+      val (results, maxScore) = repository.queryAddress(tokens).await
 
       // Then
-      result.length shouldBe 1
-      result.head shouldBe expected
+      results.length shouldBe 1
+      results.head shouldBe expected
+      maxScore shouldBe 1.4142135f
```
Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec.scala]: Make this value a `val`

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 4 above)
Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: Scaladoc is written in the parent trait

File: server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala (hunk 2 above)
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: ah. I see now. Would be better as a case class, then just `map { resp =>  //blah }`

Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: I would still write 
```
map {
  case CaseClass(address, max) => bla bla
}
```

Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: Not sure what you mean,

This is what I mean

```
esRepo.queryAddress(tokens).map { resp => 
   resp.x;
   resp.y
}

```

Not sure your example compiles..

Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: Change the return type of the future to a case class.
then use the vars in that case class.

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 3 above)
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: @Mironor  this is what i mean, generally i only return tuples when i map that function, then construct a map after. even then i dont like doing it. Return types are preferred across the board

File: server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala (hunk 2 above)
Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]:
```
case PostcodeAddressFileAddresses(addresses, maxScore) => addresses.map(AddressResponseAddress.fromPafAddress(maxScore))
```
vs
```
postcodeAddresses => postcodeAddresses.addresses.map(AddressResponseAddress.fromPafAddress(postcodeAddresses.maxScore)),
```

The first one looks better to me

Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: It's not about what looks better; there's more going on under the hood when you use the first one.

Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala]: But the first one is more readable.

**Pull request merged**